### PR TITLE
fix: tooltip

### PR DESCRIPTION
**Pull request by isabelroses**

### CODE REVIEW

**Comment by nullishamy** [comment]
Do we want to set the border colour explicitly / is it already subtext?

**Comment by isabelroses** [comment]
> Do we want to set the border colour explicitly / is it already subtext?

border seems to be set to none, `border: none;`.

**Comment by isabelroses** [comment]
I got that from this command
`cat ./dist/catppuccin-frappe-blue-standard+default-dark/gtk-4.0/gtk.css | rg tooltip -A8`

**Comment by nullishamy** [comment]
Hmm. We can always revisit that at a later date, seems good for now :+1:

**Comment by isabelroses** [comment]
Just looking into it, because it's set to none, its breaking.

**Comment by isabelroses** [comment]
Oh the patch is also failing because `mantle` and etc are never set.

File: patches/colloid/fix-tooltip.patch [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+ $drop_target_color: $accent_bg_color;
+diff --git a/src/sass/libadwaita/widgets/_tooltip.scss b/src/sass/libadwaita/widgets/_tooltip.scss
+index 2ed45776..c8450fee 100644
+--- a/src/sass/libadwaita/widgets/_tooltip.scss
++++ b/src/sass/libadwaita/widgets/_tooltip.scss
+@@ -2,12 +2,12 @@ tooltip {
+   &.background {
+     // background-color needs to be set this way otherwise it gets drawn twice
+     // see https://bugzilla.gnome.org/show_bug.cgi?id=736155 for details.
+-    background-color: transparentize(black, 0.2);
```
Comment (nullishamy) [inline patches/colloid/fix-tooltip.patch:79]: I don't think these colours are right for mocha etc. 750 is mantle, and i think we just use $white for text?

File: patches/colloid/fix-tooltip.patch [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+ $scrim-alt:                             background(g);
+diff --git a/src/sass/gtk/_common-3.0.scss b/src/sass/gtk/_common-3.0.scss
+index 0b6df5b8..ae7d0cbb 100644
+--- a/src/sass/gtk/_common-3.0.scss
++++ b/src/sass/gtk/_common-3.0.scss
+@@ -231,7 +231,7 @@ label {
+   &.osd {
+     border-radius: $corner-radius;
+     background-color: $tooltip;
+-    color: on($tooltip);
```
Comment (nullishamy) [inline patches/colloid/fix-tooltip.patch:22]: Is there a reason we can't use this?

**Review by nullishamy — CHANGES_REQUESTED** [review]

File: patches/colloid/fix-tooltip.patch (hunk 2 above)
Comment (isabelroses) [reply patches/colloid/fix-tooltip.patch:22]: the `on()` function generates a more appropriate color in this case black, so i chose to just change it.

File: patches/colloid/fix-tooltip.patch (hunk 1 above)
Comment (isabelroses) [reply patches/colloid/fix-tooltip.patch:79]: `$white` is a hardcoded value https://github.com/catppuccin/gtk/blob/4173b70b910bbb3a42ef0e329b3e98d53cef3350/patches/colloid/palette.tera#L108

Comment (nullishamy) [reply patches/colloid/fix-tooltip.patch:79]: Sure, but `white` and `$white` are two very different things, you have the former in this case? Did you mean to put the latter in?

**Review by nullishamy — CHANGES_REQUESTED** [review]

**Pull request closed without merging**